addons/compact-menu-bar/userscript.js

```diff
@@ -0,0 +1,14 @@
+export default async function ({ addon }) {
+  if (await addon.auth.fetchIsLoggedIn()) {
+    const username = await addon.auth.fetchUsername();
+    addon.tab.redux.initialize(); // Start listening to Redux events
+    addon.tab.redux.addEventListener("statechanged", async (e) => {
+      if (addon.self.disabled) return; // Skip if the addon is disabled
+      if (e.detail.action.type !== "scratch-gui/menus/OPEN_MENU" || e.detail.action.menu !== "accountMenu") return; // Skip if another Redux event occurs
+      const profileSpans = await addon.tab.waitForElement(".menu_menu-item_3EwYA.menu_hoverable_3u9dt"); // Do this if the menu is opened
+      const profile = profileSpans.appendChild(document.createElement("span")); // Add username to "Profile" menu option
```

Comment:
I don't think the addon needs to support dynamic enable with the menu is open, but could `addon.tab.displayNoneWhileDisabled` be used here so the username style doesn't get messed up when dynamically disabling the addon with the menu open?

Reply:
I'll look into it.

Reply:
Tried it now, and it messes up the menu when toggled. I think it's good enough like this.